hack/shellcheck-test.sh

```diff
@@ -15,28 +15,30 @@ test_shellcheck() {
                         return 0
                 fi
                 #shell check -x -e SC2181,SC2029,SC1091,SC1090,SC2012 "${1}"
-                shellcheck -x -e SC2181 "${1}"
+                "${SHELLCHECK}" -x -e SC2181 "${1}"
         else
                 return 0
         fi
 }
 
-SHELLCHECK="$(command -v shellcheck 2>/dev/null)"
-
 SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
 
 BASE_DIR="$(cd "${SCRIPT_DIR}/.." && pwd)"
 
-if [[ -z "${SHELLCHECK}" ]]; then
-        echo "warning: could not find shellcheck ... installing shellcheck" >&2
-        scversion="stable"
-        FILE="${OUTDIR_TOOLS}/shellcheck"
-        wget -qO- "https://storage.googleapis.com/shellcheck/shellcheck-${scversion?}.linux.x86_64.tar.xz" | tar -xJv
-        cp -f "shellcheck-${scversion}/shellcheck" "${FILE}"
-        if [ -f "$FILE" ]; then
-                SHELLCHECK="${FILE}"
-        fi
+SC_VERSION="stable"
+echo "Installing shellcheck... (${SC_VERSION?})" >&2
+FILE="${OUTDIR_TOOLS}/shellcheck"
+if [ "$OS_TYPE" == "Darwin" ]; then
+        wget -qO- "https://github.com/koalaman/shellcheck/releases/download/${SC_VERSION?}/shellcheck-${SC_VERSION?}.darwin.x86_64.tar.xz" | tar -xJv
```

Comment:
Are you unable to use the `storage.googleapis.com` version? I would hope we could simply set a variable to be either `darwin` or `linux`. How does this work by only providing `SC_VERSION`?

Reply:
The darwin version of `storage.googleapis.com` is deprecated and when you run it, it prompts you to use this one

Reply:
It seems like the github releases should be more authoritative in general. Could we use the same link for both linux and darwin?

Reply:
done

Reply:
I see you did what @jarrpa suggested: use the same base url (the github one) for both cases but did that in a later commit. Would be better to do this right away.

Also please construct the url using a conditional and then just have one occurence of the `wget | tar` command, similar to https://github.com/openshift/ocs-operator/blob/master/hack/ensure-operator-sdk.sh and https://github.com/noobaa/noobaa-operator/blob/master/build/install-operator-sdk.sh.

Reply:
I don't think it is a good way to just build the URL based on parts because we are not guaranteed to have the same URL structure for each platform. For example, up to the last commit, the Linux case was taken from google cloud and not GitHub.
But in order to not code `wget` and `tar` twice, as requested, I now save the URL (source) in a separate var then run the commands using this var. 

I think this should be an acceptable compromise.

Reply:
Well, I just wanted to highlight that @jarrpa asked you to use the same url for linux and darwin (changing the one used for linux), and you said "done" but you actually didn't do it... :wink:

Reply:
It cannot be the exact same URL in any case. 
Doing the diff part via `var` or `if` does not matter.
The main thing here is that both URLs point to Github which is more  authoritative

Reply:
I also think that er interrupt "the same URL"  as different things